drracket/help/private/bug-report-controls.rkt

```diff
@@ -1,26 +1,21 @@
 #lang racket/base
-(require racket/gui/base
+(require (for-syntax racket/base
+                     racket/list)
+         framework
+         pkg
          racket/class
          racket/contract
+         racket/gui/base
          racket/pretty
-         string-constants/string-constant
          setup/dirs
          setup/link
-         framework
-         pkg
-         (for-syntax racket/base
-                     racket/list)
+         string-constants/string-constant
          "buginfo.rkt"
          "save-bug-report.rkt")
 
-(provide/contract
- [add-bug-report-controls
-  (-> (is-a?/c area-container<%>) 
-      saved-report?
-      (-> any)
-      (-> any)
-      (-> any)
-      any)])
+(provide (contract-out
+          [add-bug-report-controls
+           (-> (is-a?/c area-container<%>) saved-report? (-> any) (-> any) (-> any) any)]))
```

Comment:
Can we ask that `(provide (contract-out` always be on two lines?

Reply:
Which "two lines" are you talking about? Is it how in each clause, the identifier should be in one line, and the corresponding contract should be in another line? That can be adjusted by changing the formatter of `contract-out`. But wouldn't this look better?

```
(provide (contract-out
          [foo (-> number? number?)]
          [bar (-> number? number?)])))
```

Or are you talking about how you always want:

```
(provide 
 (contract-out 
  [foo ...]))
```

instead?

Reply:
The latter. The `provide` and the `contract-out` on two separate lines.

It may be better to think of this as something that should happen because future lines'll have less leftward drift just in general, or may it is better to special case these two? Or maybe this is an unreasonable request? :)

Reply:
I also think it should always look like the latter.

Reply:
> Or maybe this is an unreasonable request?

It's totally reasonable! I'm simply trying to pinpoint what exactly is the general change that you want. I understand how you want the above, specific program to be formatted. But what about the below program -- which one do you prefer?

(A)

```
(provide foo
         bar
         (contract-out 
          [baz ...]))
```

vs

(B)

```
(provide 
 foo
 bar
 (contract-out 
  [baz ...]))
```

I'm mentioning this because my understanding is that:

```
(provide foo
         bar
         baz)
```

is the usual format for `provide`. So (A) seems to be more consistent with the usual format? But if that is the case, then it feels weird to me that we want:

```
(provide foo
         bar
         (contract-out 
          [baz ...]))
```

but at the same time also **don't** want

```
(provide (contract-out 
          [baz ...]))
```

One possibility to satisfy all constraints might be to have a rule like "if `contract-out` is the only subform of `provide`, then `provide` and `contract-out` should be in different lines." That would work, but I'm not sure if that's what you had in mind.

Reply:
I prefer B in the options you provided, though in not thrilled about it when it happens.

Reply:
So @jackfirth's proposed rule is that if `contract-out` is one of subforms of `provide`, then always enter a newline after `provide`. That definitely could be done if we all agree with it.

Reply:
I agree; option B is my preference too. Another option is to split the provide into two provides when there is a `contract-out` form. Maybe that's a bad idea, tho.

Reply:
Something like this? https://github.com/sorawee/fmt/pull/81

which would affect the format of:
- https://github.com/racket/racket/blob/218c28e4dfce1d3920f6079d14c19d1df68c0287/racket/collects/racket/random.rkt#L5-L10
- https://github.com/racket/racket/blob/218c28e4dfce1d3920f6079d14c19d1df68c0287/pkgs/net-lib/net/private/rr-srv.rkt#L6-L11
- https://github.com/racket/racket/blob/218c28e4dfce1d3920f6079d14c19d1df68c0287/racket/collects/racket/struct.rkt#L4-L9
- https://github.com/racket/racket/blob/218c28e4dfce1d3920f6079d14c19d1df68c0287/racket/collects/syntax/parse.rkt#L17-L26

The file `test-contract-out.rkt.out` contain some example output.